src/main/java/me/jellysquid/mods/phosphor/mixin/chunk/light/MixinSkyLightStorage.java

```diff
@@ -87,4 +89,26 @@ public int getLight(long pos) {
             }
         }
     }
+
+    @Shadow
+    protected abstract boolean isAboveTopmostLightArray(long sectionPos);
+
+    @Shadow
+    protected abstract boolean isLightEnabled(long sectionPos);
+
+    @Override
+    public int getLightWithoutLightmap(final long blockPos)
+    {
+        long sectionPos = ChunkSectionPos.offset(ChunkSectionPos.fromGlobalPos(blockPos), Direction.UP);
+
+        if (this.isAboveTopmostLightArray(sectionPos)) {
+            return this.isLightEnabled(sectionPos) ? 15 : 0;
+        }
+
+        ChunkNibbleArray lightmap;
+
+        for (; (lightmap = this.getLightArray(sectionPos, true)) == null; sectionPos = ChunkSectionPos.offset(sectionPos, Direction.UP));
```

Comment:
This for-loop would really be better replaced with a while-loop for clarity sake.

```java
while ((lightmap = this.getLightArray(sectionPos, true)) == null) {
  sectionPos = ChunkSectionPos.offset(sectionPos, Direction.UP);
}
```